test/numbers.jl

```diff
@@ -2771,7 +2771,8 @@ let types = (Base.BitInteger_types..., BigInt, Bool,
     for S in types
         for op in (+, -)
             if S === Float16 # type instability here?
-                @test_broken @inferred Base.promote_op(op, S)
+                # broken, fixme then remove this branch
```

Comment:
Maybe we should change `@test_broken` to allow the argument to either fail or throw. Or introduce another macro `@test_broken_throws`, although in that case I'd be tempted to rename these macros as `@broken_test` and `@broken_test_throws`... Or maybe better still spell them as `@broken @test` and `@broken @test_throws`, etc.